homeassistant/components/media_player/mediaroom.py

```diff
@@ -152,50 +189,112 @@ def media_content_type(self):
         """Return the content type of current playing media."""
         return MEDIA_TYPE_CHANNEL
 
-    def turn_on(self):
-        """Turn on the receiver."""
-        self.stb.send_cmd('Power')
-        self._state = STATE_ON
+    @property
+    def media_channel(self):
+        """Channel currently playing."""
+        return self._channel
 
-    def turn_off(self):
+    async def async_turn_on(self):
+        """Turn on the receiver."""
+        from pymediaroom import PyMediaroomError
+        try:
+            self.set_state(await self.stb.turn_on())
+            if self._optimistic:
+                self._state = STATE_PLAYING
+        except PyMediaroomError:
+            self._available = False
```

Comment:
You will need to set `self._available = True` inside your `try` and after the command has been successful, otherwise a device will never be marked as available again if it reconnects.